patroni/postgresql.py

```diff
@@ -448,11 +448,14 @@ def check_replication_lag(self, last_leader_operation):
 
     def write_pg_hba(self):
         with open(os.path.join(self.data_dir, 'pg_hba.conf'), 'a') as f:
-            f.write('\nhost replication {username} {network} md5\n'.format(**self.replication))
-            for line in self.config.get('pg_hba', []):
-                if line.split()[0].strip() == 'hostssl' and self.server_parameters.get('ssl', 'off').lower() != 'on':
-                    continue
-                f.write(line + '\n')
+            ssl = self.server_parameters.get('ssl', 'off').lower() == 'on'
+
+            replication_record = '{method} replication {username} {network} md5'.format(
+                                 method='hostssl' if ssl else 'host', **self.replication)
+
+            for line in ['', replication_record] + self.config.get('pg_hba', []):
+                if ssl or not line.startswith('hostssl'):
```

Comment:
Oh, it's the other way round: it will reject user-supplied `hostssl` if we don't have SSL enabled.  Any reason to filter this explicitly?  I'd just let it go to the config, it won't work either way.